code/weapon/weapons.cpp

```diff
@@ -5106,6 +5139,21 @@ int weapon_create( vec3d * pos, matrix * porient, int weapon_type, int parent_ob
 		}
 	}
 
+	// Let's setup a fast failure check with a uniform distribution.
+	if (wip->failure_rate > 0.0f) {
+		std::random_device rd;
+		std::mt19937 f(rd());
+		std::uniform_real_distribution<float> gen(0.0f, 1.0f);
```

Comment:
You should use `UniformFloatRange` here since that does exactly what you need without all the additional code.

Reply:
It's in `utils/RandomRange.h`.